From 2dd60a2d131ab3b6d608ac4d6e09a9ed35c9a640 Mon Sep 17 00:00:00 2001
From: Damian Hobson-Garcia <dhobsong@igel.co.jp>
Date: Tue, 24 Jul 2012 11:13:52 +0900
Subject: [PATCH 08/39] Re-initialize OMX parameter structure before re-use

The OMX_PARAM_PORTDEFINITIONTYPE structure is 0-cleared between setting the
input and output port properties.  After being cleared the structure needs
to be re-initialized before it can be used again.
---
 omx/gstomx_base_videodec.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/omx/gstomx_base_videodec.c b/omx/gstomx_base_videodec.c
index d5fdc38..b41f17f 100644
--- a/omx/gstomx_base_videodec.c
+++ b/omx/gstomx_base_videodec.c
@@ -199,9 +199,11 @@ omx_setup (GstOmxBaseFilter * omx_base)
       OMX_SetParameter (gomx->omx_handle, OMX_IndexParamPortDefinition, &param);
     }
 
+    memset (&param, 0, sizeof (param));
+    G_OMX_INIT_PARAM (param);
+
     /* Output port configuration */
     {
-      memset (&param, 0, sizeof (param));
       param.nSize = sizeof (OMX_PARAM_PORTDEFINITIONTYPE);
 
       param.nPortIndex = 1;
-- 
1.7.9.5

